packages/strapi-plugin-upload/middlewares/upload/index.js

```diff
@@ -1,15 +1,16 @@
 'use strict';
 
-const { join } = require('path');
+const { resolve } = require('path');
 const range = require('koa-range');
 const koaStatic = require('koa-static');
 
 module.exports = strapi => ({
   initialize() {
-    const staticDir = join(
-      strapi.dir,
-      strapi.config.middleware.settings.public.path || strapi.config.paths.static
+    const configPublicPath = strapi.config.get(
+      'middleware.settings.public.path',
+      strapi.config.paths.static
     );
+    const staticDir = resolve(configPublicPath);
```

Comment:
you need to call resolve(strapi.dir, publicPath) or this will not work with relative path